iothub_client/inc/blob.h

```diff
@@ -44,10 +45,11 @@ DEFINE_ENUM(BLOB_RESULT, BLOB_RESULT_VALUES)
 * @param    httpStatus      A pointer to an out argument receiving the HTTP status (available only when the return value is BLOB_OK)
 * @param    httpResponse    A BUFFER_HANDLE that receives the HTTP response from the server (available only when the return value is BLOB_OK)
 * @param    certificates    A null terminated string containing CA certificates to be used
+* @param    proxyOptions    A structure that contains optional web proxy information
 *
 * @return	A @c BLOB_RESULT. BLOB_OK means the blob has been uploaded successfully. Any other value indicates an error
 */
-MOCKABLE_FUNCTION(, BLOB_RESULT, Blob_UploadFromSasUri,const char*, SASURI, const unsigned char*, source, size_t, size, unsigned int*, httpStatus, BUFFER_HANDLE, httpResponse, const char*, certificates)
+MOCKABLE_FUNCTION(, BLOB_RESULT, Blob_UploadFromSasUri,const char*, SASURI, const unsigned char*, source, size_t, size, unsigned int*, httpStatus, BUFFER_HANDLE, httpResponse, const char*, certificates, HTTP_PROXY_OPTIONS*, proxyOptions)
```

Comment:
>HTTP_PROXY_OPTIONS*, proxyOptions [](start = 203, length = 33)

Can we look at passing this thru SetOption()?

Reply:
You do realize that this value comes from a SetOption call at the layer above? Unfortunately the options are not neatly packaged in a manner that would make it easy to pass them to this code. It would create a dependency on the layout of what is effectively an opaque structure.

---
In reply to: [149761106](https://github.com/Azure/azure-iot-sdk-c/pull/267#discussion_r149761106) [](ancestors = 149761106)